Editor/AGXUnityEditor/Windows/ConvertPhysXToAGXWindow.cs

```diff
@@ -0,0 +1,917 @@
+      }
+
+      using ( new UndoCollapseBlock( "Convert Selected Prefab Assets" ) ) {
+        foreach ( var prefab in m_prefabPhysXAssets ) {
+          if ( !prefab.Convert )
+            continue;
+
+          GameObject prefabObject = PrefabUtility.LoadPrefabContents(prefab.Path);
+          if ( prefabObject == null )
+            continue;
+
+          var objects = GatherObjects(prefabObject);
+          try {
+            ConvertObjects( objects );
+
+            PrefabUtility.SaveAsPrefabAsset(prefabObject, prefab.Path);
+          }
+          finally {
+            PrefabUtility.UnloadPrefabContents( prefabObject );
+          }
+
+          Debug.Log( $"Converted prefab: {prefab.Name}, objects: {objects.Count}" );
+        }
+      }
+
+      AssetDatabase.SaveAssets();
+      Repopulate( false );
+    }
+
+    private void ConvertRigidbody( AssetData asset )
+    {
+      var physXRb = asset.gameObject.GetComponent<Rigidbody>();
+      var agxRB = Undo.AddComponent<AGXUnity.RigidBody>(asset.gameObject);
+
+      agxRB.MassProperties.Mass.UseDefault = false;
+      agxRB.MassProperties.Mass.Value = physXRb.mass;
+
+#if UNITY_6000_0_OR_NEWER
+      var linearVelocityDamping = physXRb.linearDamping;
+      var angularVelocityDamping = physXRb.angularDamping;
+#else
+      var linearVelocityDamping = physXRb.drag;
+      var angularVelocityDamping = physXRb.angularDrag;
+#endif
+      agxRB.LinearVelocityDamping = new Vector3( linearVelocityDamping, linearVelocityDamping, linearVelocityDamping );
+      agxRB.AngularVelocityDamping = new Vector3( angularVelocityDamping, angularVelocityDamping, angularVelocityDamping );
+
+      if ( !physXRb.automaticCenterOfMass ) {
+        agxRB.MassProperties.CenterOfMassOffset.UseDefault = false;
+        agxRB.MassProperties.CenterOfMassOffset.Value = physXRb.centerOfMass;
+      }
+
+      if ( !physXRb.automaticInertiaTensor ) {
+        agxRB.MassProperties.InertiaDiagonal.UseDefault = false;
+        agxRB.MassProperties.InertiaDiagonal.Value = physXRb.centerOfMass;
+      }
+
+      if ( physXRb.isKinematic )
+        agxRB.MotionControl = agx.RigidBody.MotionControl.KINEMATICS;
+
+      if ( asset.rootAsset )
+        asset.gameObject.AddComponent<AGXUnity.ArticulatedRoot>();
+
+      Undo.DestroyObjectImmediate( physXRb );
+    }
+
+    private void ConvertCollider( AssetData asset )
+    {
+      // We need to create a child gameObject with the AGX collider to account for the "Center" property
+      var newObject = new GameObject();
+
+      Undo.RegisterCreatedObjectUndo( newObject, newObject.name );
+      newObject.transform.parent = asset.gameObject.transform;
+
+      newObject.transform.localRotation = Quaternion.identity;
+      newObject.transform.localScale = Vector3.one; // AGX colliders ignore transform scale, except MeshColliders
+
+      newObject.isStatic = asset.gameObject.isStatic;
+
+      var colliderScale = asset.gameObject.transform.lossyScale;
```

Comment:
It's not clear to me that this works in all cases, have you tried with nested scales and/or rotated objects at different nesting levels?